src/crucible.rs

```diff
@@ -98,27 +112,53 @@ impl Crucible {
 
     pub fn action(&mut self, crucible_action: CrucibleAction) {
         match crucible_action {
-            CrucibleAction::BakeBrick(brick_index) => {
+            BakeBrick(brick_index) => {
                 let oven = Oven::new(brick_index);
                 self.fabric = oven.prototype_fabric();
                 self.stage = BakingBrick(oven);
             }
-            CrucibleAction::BuildFabric(fabric_plan) => {
+            BuildFabric(fabric_plan) => {
                 self.fabric = Fabric::default_bow_tie();
-                self.frozen_fabric = None;
                 self.stage = RunningPlan(PlanRunner::new(fabric_plan));
             }
-            CrucibleAction::CreateBrickOnFace(face_id) => {
+            ProposeBrick(brick_on_face) => {
                 let Tinkering(tinkerer) = &mut self.stage else {
                     panic!("cannot add brick unless tinkering");
                 };
-                let spin = self.fabric.face(face_id).spin.opposite();
-                let face_alias = FaceAlias::single("Single") + &spin.into_alias();
-                tinkerer.add_brick(face_alias, face_id);
+                tinkerer.propose_brick(brick_on_face);
             }
-            CrucibleAction::SetSpeed(iterations_per_frame) => {
+            ConnectBrick => {
+                let Tinkering(tinkerer) = &mut self.stage else {
+                    panic!("cannot add brick unless tinkering");
```

Comment:
Can this be an unreachable!()?

Reply:
I've now delegated the tinkerer actions all in one.